README.md

```diff
@@ -1559,28 +1256,357 @@ let main ~config_changed =
 
 See the `examples/signals` directory for the full code.
 
-## Design Note: Determinism
+## Multicore Support
 
-Within a domain, fibers are scheduled deterministically.
-Programs using only the Eio APIs can only behave non-deterministically if given a capability to do so from somewhere else.
+Fibers are scheduled cooperatively within a single domain, but you can also create new domains that run in parallel.
+This is useful to perform CPU-intensive operations quickly.
 
-For example, `Fiber.both f g` always starts running `f` first,
-and only switches to `g` when `f` finishes or performs an effect that can switch fibers.
+There are 2 main ways of achieving this and both are documented below:
+- `Eio.Executor_pool`
+- spawning a one-off domain using `Eio.Domain_manager.run`
 
-Performing IO with external objects (e.g., `stdout`, files, or network sockets) will introduce non-determinism,
-as will using multiple domains.
+The Executor Pool is the recommended way of leveraging OCaml 5's multicore capabilities.
+It's built on top of `Eio.Domain_manager` and offers several advantages:
+- Executor Pool domains (threads) are initialized in advance to minimize overhead. They're also reused, unlike direct usage of `Eio.Domain_manager.run`.
+- Each Executor Pool domain worker can execute multiple jobs concurrently. Whenever a job blocks due to e.g. I/O, the domain's scheduler switches execution to another fiber and/or job.
+- The Executor Pool supports heterogenous jobs efficiently. For example, it will seamlessly run a large number of I/O-bound jobs on one domain while reserving an entire other domain (thread) for a single CPU-bound job. You control this behavior through the `~weight` argument required to submit a job.
```

Comment:
Maybe it's worth mentioning `weight` allows running computations in a domain until it reaches capacity. I understood this after looking at the mli file.

I'm also curious whether we can automate this somehow. There's a risk of users providing inaccurate weights. But that's not a conversation for this PR.

Reply:
Good point, I've made a few changes to help.

I think it's okay if the average user ends up looking at the mli through their code editor or on the https://ocaml-multicore.github.io/eio/eio/Eio/ site once they start using a module if it helps keep the README more focused on just: the purpose of a module, how it fits within the library, and advice about using it effectively.

> I'm also curious whether we can automate this somehow. There's a risk of users providing inaccurate weights. But that's not a conversation for this PR.

I've been thinking the same thing but I haven't been able to figure out a decent abstraction that brings enough value to justify losing the simplicity (in the API) of a bare float. But yeah, that's a separate discussion for another time and place. Please let me know if you come up with any ideas or even fragments of ideas!